plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/ForLazyConnectionFactory.java

```diff
@@ -1,31 +0,0 @@
-/*
```

Comment:
" Simplify ConnectionFactory chain of responsibility "

Does this actually simplify? Having explicit binding annotation for each "layer" is useful because then you can look at each method signature and as long as an annotation isn't used in two constructors you can be sure there isn't any duplication going on.

e.g. before this change the stack was:

```
CachingJdbcClient#ctor
JdbcDiagnosticsModule#createJdbcClientWithStats == @StatsCollecting
ClientModule#setup (OracleClient etc.) == @ForBaseJdbc
```

AND

```
ReusableConnectionFactoryModule#getReusableConnectionFactory (if reusable connections are enabled)
LazyConnectionFactory#ctor (if reusable connections is disabled)
JdbcModule#setup == @ForLazyConnectionFactory
JdbcDiagnosticsModule#createConnectionFactoryWithStats == @StatsCollecting
ClientModule#createConnectionFactory (OracleClientModule etc.) == @ForBaseJdbc
```

After this change at the very least `StatisticsAwareConnectionFactory` is duplicated in the stack.

While here i eyeballed the stack another option is to throw/add breakpoint from some method at the leaf class and then see resulting stacktrace.

Reply:
ignore most of the above comment.

the only relevant change is that now the `StatisticsAwareConnectionFactory` wraps the `DriverConnectionFactory` directly instead of wrapping `RetryingConnectionFactory` too which means that any stats from retry operations won't be accounted for anymore. Might be easy to resolve by changing what class is passed in to which constructor.

@kokosing is this something to consider?

Reply:
This is expected and desired. Before multiple attempts were counted as single connection open. Now each attempt is a separate metric.